server/sonar-db-dao/src/main/java/org/sonar/db/component/BranchDao.java

```diff
@@ -49,6 +49,15 @@ public void upsert(DbSession dbSession, BranchDto dto) {
     }
   }
 
+  public void updateMainBranchName(DbSession dbSession, String projectUuid, String newBranchKey) {
```

Comment:
ComponentKeyUpdaterDao is a bad example as it should be refactored :(.
If you don't want to use upsert, then create an update method